test/gtest/aca_test_dhcp.cpp

```diff
@@ -180,3 +224,341 @@ TEST(dhcp_message_test_cases, get_options_valid)
+  string cmd_string;
+  int overall_rc;
+
+  // delete br-int and br-tun bridges
+  ACA_OVS_L2_Programmer::get_instance().execute_ovsdb_command(
+          "del-br br-int", not_care_culminative_time, overall_rc);
+
+  ACA_OVS_L2_Programmer::get_instance().execute_ovsdb_command(
+          "del-br br-tun", not_care_culminative_time, overall_rc);
+
+  // create and setup br-int and br-tun bridges, and their patch ports
+  overall_rc = ACA_OVS_L2_Programmer::get_instance().setup_ovs_bridges_if_need();
+  ASSERT_EQ(overall_rc, EXIT_SUCCESS);
+
+  GoalState GoalState_builder;
+  DHCPState *new_dhcp_states = GoalState_builder.add_dhcp_states();
+  SubnetState *new_subnet_states = GoalState_builder.add_subnet_states();
+
+  new_dhcp_states->set_operation_type(OperationType::CREATE);
+
+  // fill in port state structs
+  DHCPConfiguration *DHCPConfiguration_builder = new_dhcp_states->mutable_configuration();
+  DHCPConfiguration_builder->set_revision_number(1);
+  DHCPConfiguration_builder->set_message_type(MessageType::FULL);
+
+  DHCPConfiguration_builder->set_subnet_id(subnet_id_1);
+  DHCPConfiguration_builder->set_mac_address(vmac_address_1);
+  DHCPConfiguration_builder->set_ipv4_address(vip_address_1);
+  DHCPConfiguration_builder->set_port_host_name(port_name_1);
+
+  // dns entry
+  DHCPConfiguration_DnsEntry *DnsEntry_builder = DHCPConfiguration_builder->add_dns_entry_list();
+  DnsEntry_builder->set_entry(port_dns_entry);
+
+  // fill in subnet state structs
+  SubnetConfiguration *SubnetConiguration_builder =
+          new_subnet_states->mutable_configuration();
+  SubnetConiguration_builder->set_revision_number(1);
+  SubnetConiguration_builder->set_project_id(project_id);
+  SubnetConiguration_builder->set_vpc_id(vpc_id_1);
+  SubnetConiguration_builder->set_id(subnet_id_1);
+  SubnetConiguration_builder->set_cidr(subnet1_cidr);
+  SubnetConiguration_builder->set_tunnel_id(123);
+
+  auto *subnetConfig_GatewayBuilder(new SubnetConfiguration_Gateway);
+  subnetConfig_GatewayBuilder->set_ip_address(subnet1_gw_ip);
+  subnetConfig_GatewayBuilder->set_mac_address(subnet1_gw_mac);
+  SubnetConiguration_builder->set_allocated_gateway(subnetConfig_GatewayBuilder);
+
+  // create a new port 1 in demo mode
+  GoalStateOperationReply gsOperationalReply;
+
+  overall_rc = Aca_Comm_Manager::get_instance().update_goal_state(
+          GoalState_builder, gsOperationalReply);
+  ASSERT_EQ(overall_rc, EXIT_SUCCESS);
+
+  // create a new port 2 in demo mode
+  DHCPConfiguration_builder->set_mac_address(vmac_address_2);
+  DHCPConfiguration_builder->set_ipv4_address(vip_address_2);
+  DHCPConfiguration_builder->set_port_host_name(port_name_2);
+
+  overall_rc = Aca_Comm_Manager::get_instance().update_goal_state(
+          GoalState_builder, gsOperationalReply);
+  ASSERT_EQ(overall_rc, EXIT_SUCCESS);
+
+  // send goal state ok, now create a docker instance test
+  // con1
+  overall_rc = Aca_Net_Config::get_instance().execute_system_command(
```

Comment:
since running docker is only available in physical machine or VM. Not inside a container as the default build environment. Please go ahead to marked this test as disabled by default:

TEST(dhcp_request_test_case, DISABLED_l2_dhcp_test)

We are planning to setup a test framework to run these disabled tests that require physical machine or to enable multi machine traffic tests.